### ISSUE TITLE
Python error with GC3Pie

### ISSUE THREAD

**Issue by verdurin** [hpcugent/easybuild-easyconfigs#3295]
Trying to build `GC3Pie` on a CentOS 7.2 host:

```
configure: creating ./config.status
config.status: creating src/config.h
warning: GMP or MPIR library not found; Not building Crypto.PublicKey._fastmath.
building 'Crypto.Hash._MD2' extension
creating build/temp.linux-x86_64-2.7
creating build/temp.linux-x86_64-2.7/src
gcc -pthread -fno-strict-aliasing -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -m64 -mtune=generic -D_GNU_SOURCE -fPIC -fwrapv -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -m64 -mtune=generic -D_GNU_SOURCE -fPIC -fwrapv -fPIC -std=c99 -O3 -fomit-frame-pointer -Isrc/ -I/usr/include/python2.7 -c src/MD2.c -o build/temp.linux-x86_64-2.7/src/MD2.o
src/MD2.c:31:20: fatal error: Python.h: No such file or directory
 #include "Python.h"
                    ^
compilation terminated.
error: command 'gcc' failed with exit status 1
```

**Comment by verdurin** [comment]
As you'd expect, installing `python-devel` fixes this. 

Ought that to be included explicitly as an OS dep?

**Comment by boegel** [comment]
@verdurin I guess so, yes.

@riccardomurri thoughts?

**Comment by riccardomurri** [comment]
Well that depends on what Python interpreter is being used.

If GC3Pie is going to be run using the system Python then yes, either `python-devel` (RHEL/CentOS) or `python-dev` (Debian/Ubuntu) should be installed.

If using an EB-installed Python then I guess `Python.h` should be somewhere among the installed files?

**Comment by boegel** [comment]
This concerns the system Python, so yes, in that case, we should include this in the GC3Pie easyconfigs:

```
osdependencies = [('python-devel', 'python-dev')]
```

When an EasyBuild-installed Python is installed, there's indeed no issue, since we don't distinguish between `python` and `python-devel`.

**Comment by verdurin** [comment]
I'm agnostic as to whether system or EB Python is used, but the dependency needs to be included in the easyconfig.

**Comment by boegel** [comment]
fixed in #3310
